services/worker-manager/src/provisioner.js

```diff
@@ -161,76 +165,73 @@ class Provisioner {
       const poolsByProvider = new Map();
 
       // Now for each worker pool we ask the providers to do stuff
-      await this.WorkerPool.scan({}, {
-        handler: async workerPool => {
-          const {providerId, previousProviderIds, workerPoolId} = workerPool;
-          const provider = this.providers.get(providerId);
+      const workerPools = (await this.db.fns.get_worker_pools(null, null)).map(row => WorkerPool.fromDb(row));
```

Comment:
It could, but I bet that would be thousands of worker pools.  I think this is something we could double back on later, perhaps when we've added pagination based on keys (as you've suggested) and written some kind of postgres-to-async-iterator utility.

Reply:
>  I think this is something we could double back on later, perhaps when we've added pagination based on keys

I'm not sure I understand. Doesn't `get_worker_pools` already support pagination?

> and written some kind of postgres-to-async-iterator utility

Could you remind me what was the benefit of adding this utility, please?

Reply:
It does, but using that pagination is a bit complicated -- enough that it would need additional testing.  I'd like to write a utility similar to paginateResults but for usage within the services that builds an AsyncIterator from a paginated DB call.